tldr/change_impact.py

```
@@ -221,11 +242,19 @@ def collect_test_files(node: dict):
+                collect_test_files(target_node)
 
         except Exception:
             # If impact analysis fails for a function, continue
-            pass
+            logger.debug("Failed to analyze impact for function %s", func_name, exc_info=True)
+
+    # Scan project files once and reuse for import search + test count
+    all_files = []
+    try:
+        all_files = scan_project_files(str(project), language=language)
+    except Exception:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)